MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,224 @@
+
+            string includePattern = null;
+            string excludePattern = null;
+            var usings = new List<string>();
+
+            Exception MissingArgValue() =>
+                new InvalidOperationException("Missing argument value.");
+
+            using (var arg = args.GetEnumerator())
+            {
+                while (arg.MoveNext())
+                {
+                    switch (arg.Current)
+                    {
+                        case "-i":
+                        case "--include":
+                            includePattern = Read(arg, MissingArgValue);
+                            break;
+                        case "-x":
+                        case "--exclude":
+                            excludePattern = Read(arg, MissingArgValue);
+                            break;
+                        case "-u":
+                        case "--using":
+                            usings.Add(Read(arg, MissingArgValue));
+                            break;
+                        case "":
+                            continue;
+                        default:
+                            dir = arg.Current[0] != '-'
+                                ? arg.Current
+                                : throw new Exception("Invalid argument: " + arg.Current);
+                            break;
+                    }
+                }
+            }
+
+            var includePredicate
+                = string.IsNullOrEmpty(includePattern)
+                ? (_ => true)
+                : new Func<string, bool>(new Regex(includePattern).IsMatch);
+
+            var excludePredicate
+                = string.IsNullOrEmpty(excludePattern)
+                ? (_ => false)
+                : new Func<string, bool>(new Regex(excludePattern).IsMatch);
+
+            var thisAssemblyName = typeof(Program).GetTypeInfo().Assembly.GetName();
+
+            var q =
+                from fp in Directory.EnumerateFiles(dir, "*.cs")
+                select new
+                {
+                    SourcePath = fp,
+                    Methods =
+                        from cd in
+                            CSharpSyntaxTree
+                                .ParseText(File.ReadAllText(fp), CSharpParseOptions.Default.WithPreprocessorSymbols("MORELINQ"))
+                                .GetRoot()
+                                .SyntaxTree
+                                .GetCompilationUnitRoot()
+                                .DescendantNodes().OfType<ClassDeclarationSyntax>()
+                        where (string) cd.Identifier.Value == "MoreEnumerable"
+                        from md in cd.DescendantNodes().OfType<MethodDeclarationSyntax>()
+                        let mn = (string) md.Identifier.Value
+                        where md.ParameterList.Parameters.Count > 0
+                        && md.ParameterList.Parameters.First().Modifiers.Any(m => (string)m.Value == "this")
```

Comment:
This leaves non-extension methods out of the new namespace. This will have the unfortunate consequence of easily causing conflicts, as the `MoreLinq` namespace will need to be imported to use `Generate` or `Random`.